### PR TITLE
[Dynamo] SizeVariable can be indexed by symint

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #ISSUE_NUMBER

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/dynamo/test_functions.py [hunk 1]
```diff
@@ -1196,6 +1196,16 @@ def test_partials_lambda(x):
...
+    def test_tensor_size_indexed_by_symint(self):
+        def fn(x, y):
+            index = x.shape[-1]
+            return x + y.shape[index]
+
+        x = torch.rand(10, 2)
+        y = torch.rand(10, 8, 6)
+        opt_fn = torch.compile(backend="eager", fullgraph=True)(fn)
+        self.assertEqual(opt_fn(x, y), fn(x, y))
+
```
Comment (yanboliang) [inline test/dynamo/test_functions.py:1208]: The dynamic shape version test would be generated by ```test/dynamo/test_dynamic_shapes.py``` automatically.

**Review by williamwen42 — APPROVED** [review]

**Pull request closed without merging**